core/shared/src/main/scala/zio/prelude/AssociativeEither.scala

```diff
@@ -218,7 +218,7 @@ object AssociativeEither {
   implicit def ZLayerAssociativeEither[R, E]: AssociativeEither[({ type lambda[+a] = ZLayer[R, E, a] })#lambda] =
     new AssociativeEither[({ type lambda[+a] = ZLayer[R, E, a] })#lambda] {
       def either[A, B](fa: => ZLayer[R, E, A], fb: => ZLayer[R, E, B]): ZLayer[R, E, Either[A, B]] =
-        fa.map(Left(_)) orElse fb.map(Right(_))
+        fa.map(e => ZEnvironment(Left(e.get[A]))) orElse fb.map(e => ZEnvironment(Right(e.get[B])))
```

Comment:
This gives
```
zio-prelude/core/shared/src/main/scala/zio/prelude/AssociativeEither.scala:221:44: Cannot find implicit for IsNotIntersection[A]
[error]         fa.map(e => ZEnvironment(Left(e.get[A]))) orElse fb.map(e => ZEnvironment(Right(e.get[B])))
[error]                                            ^
```

Reply:
This is not going to be well defined anymore because conceptually you could say that give a layer that produces `ZEnvironment[A]` and a layer that produces `ZEnvironment[B]` we can either produce `ZEnvironment[A]` or `ZEnvironment[B]`. But the actual signature says we can produce a `ZEnvironment[Either[A, B]]` which indicates an environment containing a single either `A` or `B` rather than an environment containing potentially a whole set of services `A` or an environment containing a whole different set of services `B`.